data/src/main/java/com/teamagam/gimelgimel/data/images/ImageUtils.java

```diff
@@ -34,36 +46,26 @@ public static Uri getTempImageUri(ExternalDirProvider externalDirProvider){
         return Uri.fromFile(tempFile);
     }
 
-    public static byte[] readAndCompressImage(File image) {
-        BitmapFactory.Options options = new BitmapFactory.Options();
-        options.inJustDecodeBounds = true;
-        BitmapFactory.decodeFile(image.getPath(), options);
-
-        int scale = getScaleSize(options.outHeight, options.outWidth);
+    public byte[] readAndCompressImage(File image) {
+        File compressorImage = new Compressor.Builder(mContext)
+                .setCompressFormat(Constants.IMAGE_COMPRESS_TYPE)
+                .setMaxHeight(Constants.COMPRESSED_IMAGE_MAX_DIMENSION_PIXELS)
+                .setMaxWidth(Constants.COMPRESSED_IMAGE_MAX_DIMENSION_PIXELS)
+                .setQuality(Constants.COMPRESSED_IMAGE_JPEG_QUALITY)
+                .build().compressToFile(image);
 
-        //Decode with inSampleSize
-        options = new BitmapFactory.Options();
-        options.inSampleSize = scale;
-
-        ByteArrayOutputStream stream = new ByteArrayOutputStream();
-        Bitmap compressedBitmap = BitmapFactory.decodeFile(image.getPath(), options);
-
-        compressedBitmap.compress(Constants.IMAGE_COMPRESS_TYPE, Constants.COMPRESSED_IMAGE_JPEG_QUALITY, stream);
-
-        return stream.toByteArray();
+        return getImageBytes(compressorImage);
     }
 
-    private static int getScaleSize(int outHeight, int outWidth) {
-        int scale = 1;
-
-        if (outHeight > Constants.COMPRESSED_IMAGE_MAX_DIMENSION_PIXELS ||
-                outWidth > Constants.COMPRESSED_IMAGE_MAX_DIMENSION_PIXELS) {
-            scale = (int) Math.pow(2,
-                    (int) Math.ceil(Math.log(Constants.COMPRESSED_IMAGE_MAX_DIMENSION_PIXELS /
-                            (double) Math.max(outHeight, outWidth)) / Math.log(0.5)
-                    ));
+    private byte[] getImageBytes(File image) {
+        try {
+            byte[] compressedImageBytes = new byte[(int) image.length()];
+            FileInputStream stream = new FileInputStream(image.getAbsolutePath());
+            stream.read(compressedImageBytes);
+            stream.close();
```

Comment:
closing stream should be done at finally block (there's a small problem where the close method itself can throw an exception and therefore it should be taken care of aswell).
You can read about it here - 
http://stackoverflow.com/questions/2699209/java-io-ugly-try-finally-block